flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FileCatalogStore.java

```diff
@@ -0,0 +1,216 @@
+
+    /** The directory path where catalog configurations will be stored. */
+    private final String catalogStoreDirectory;
+
+    /** The character set to use when reading and writing catalog files. */
+    private final String charset;
+
+    /** A map of catalog file names to their corresponding paths. */
+    private final Map<String, Path> files = new ConcurrentHashMap<>();
+
+    /** The YAML parser to use when reading and writing catalog files. */
+    private final Yaml yaml = new Yaml();
+
+    /**
+     * Creates a new {@link FileCatalogStore} instance with the specified directory path.
+     *
+     * @param catalogStoreDirectory the directory path where catalog configurations will be stored
+     */
+    public FileCatalogStore(String catalogStoreDirectory, String charset) {
+        this.catalogStoreDirectory = catalogStoreDirectory;
+        this.charset = charset;
+    }
+
+    /**
+     * Opens the catalog store and initializes the catalog file map.
+     *
+     * @throws CatalogException if the catalog store directory does not exist or if there is an
+     *     error reading the directory
+     */
+    @Override
+    public void open() throws CatalogException {
+        super.open();
+
+        try {
+            File directoryFile = new File(catalogStoreDirectory);
+            if (!directoryFile.isDirectory()) {
+                throw new CatalogException("Catalog store only support local directory.");
+            }
+            Files.list(directoryFile.toPath())
+                    .filter(file -> file.getFileName().toString().endsWith(FILE_EXTENSION))
+                    .filter(Files::isRegularFile)
+                    .forEach(
+                            p ->
+                                    files.put(
+                                            p.getFileName().toString().replace(FILE_EXTENSION, ""),
+                                            p));
+        } catch (IOException e) {
+            throw new CatalogException("Failed to open catalog store directory", e);
+        }
+    }
+
+    /**
+     * Stores the specified catalog in the catalog store.
+     *
+     * @param catalogName the name of the catalog
+     * @param catalog the catalog descriptor to store
+     * @throws CatalogException if the catalog store is not open or if there is an error storing the
+     *     catalog
+     */
+    @Override
+    public void storeCatalog(String catalogName, CatalogDescriptor catalog)
+            throws CatalogException {
+        checkOpenState();
+
+        Path filePath = files.computeIfAbsent(catalogName, name -> getCatalogPath(catalogName));
+
+        try {
+            String yamlString = yaml.dumpAsMap(catalog.getConfiguration().toMap());
+            filePath.toFile().createNewFile();
+            FileUtils.writeFile(filePath.toFile(), yamlString, charset);
+            LOG.info("Catalog %s configuration saved to file: {}", filePath);
+        } catch (IOException e) {
```

Comment:
We can use `Throwable` to try to catch all possible  exception, e.g `path.toFile()` may throw `UnsupportedOperationException`